### PR TITLE
Free memory leaked by Darwin's Core Foundation

### PR DESCRIPTION
**Pull request by AtariDreams**
There is an unlikely but still possible edge case where factories is leaked if the array being returned is not NULL but has a count of 0. This is because the CFRelease is at the end of the if statement brackets, when it should be at the end of the plugin bracket.

### CODE REVIEW

**Comment by AtariDreams** [comment]
@zdohnal @michaelrsweet Thoughts on this?

**Comment by michaelrsweet** [comment]
@AtariDreams I've been out on vacation, but regardless please don't assume we can review and respond to PRs immediately, particularly for macOS code changes that aren't being picked up by Apple... :/

File: backend/usb-darwin.c [hunk 1]
```diff
@@ -1885,16 +1885,23 @@ static kern_return_t registry_close(void)
 static CFStringRef copy_value_for_key(CFStringRef deviceID,
 				      CFStringRef *keys)
 {
-  CFStringRef	value = NULL;
-  CFArrayRef	kvPairs = deviceID != NULL ? CFStringCreateArrayBySeparatingStrings(NULL, deviceID, CFSTR(";")) : NULL;
-  CFIndex	max = kvPairs != NULL ? CFArrayGetCount(kvPairs) : 0;
-  CFIndex	idx = 0;
+  CFStringRef value;
```
Comment (michaelrsweet) [inline backend/usb-darwin.c:1888]: Add comments

File: backend/usb-darwin.c [hunk 2]
```diff
@@ -2047,25 +2057,26 @@ static void parse_options(char *options,
 static void setup_cfLanguage(void)
 {
   CFStringRef	lang[1] = {NULL};
-  CFArrayRef	langArray = NULL;
-  const char	*requestedLang = NULL;
+  CFArrayRef	langArray;
```
Comment (michaelrsweet) [inline backend/usb-darwin.c:2060]: Comments

File: cups/tls-darwin.c [hunk 3]
```diff
@@ -1792,12 +1791,12 @@ _httpTLSStop(http_t *http)		/* I - HTTP connection */
     usleep(1000);
 
   CFRelease(http->tls);
+  http->tls = NULL;
 
-  if (http->tls_credentials)
+  if (http->tls_credentials) {
```
Comment (michaelrsweet) [inline cups/tls-darwin.c:1796]: coding style

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Some minor changes needed.

**Comment by AtariDreams** [comment]
@michaelrsweet Done!

**Review by michaelrsweet — APPROVED** [review]

**Pull request merged**